tests/preprocessing/test_positionfixes.py

```diff
@@ -1,116 +1,225 @@
+        generation using the 'between_staypoints' method with positionfixes with 'staypoint_id' and staypoints as
+        input (case 2).
+        """
+        pfs, spts = geolife_pfs_spts_short
+        pfs, tpls = pfs.as_positionfixes.generate_triplegs(spts, method='between_staypoints')
+
+        # check if staypoint/tripleg ids are present
+        assert (pfs.geometry.geom_almost_equals(testdata_pfs_ids_geolife.geometry)).all()
+        pd.testing.assert_frame_equal(pfs.drop('geom', axis=1), testdata_pfs_ids_geolife.drop('geom', axis=1),
+                                      check_like=True)
+
+    def test_tripleg_ids_in_positionfixes_case2(self, geolife_pfs_spts_short, testdata_pfs_ids_geolife):
+        """
+        checks if positionfixes are not altered and if staypoint/tripleg ids where set correctly during tripleg
+        generation using the 'between_staypoints' method with positionfixes without 'staypoint_id' and staypoints as
+        input (case 2).
+        """
+        pfs, spts = geolife_pfs_spts_short
+        pfs.drop('staypoint_id', axis=1, inplace=True)
+        pfs, tpls = pfs.as_positionfixes.generate_triplegs(spts, method='between_staypoints')
+
+        testdata_pfs_ids_geolife.drop('staypoint_id', axis=1, inplace=True)
+
+        # check if staypoint/tripleg ids are present
+        assert (pfs.geometry.geom_almost_equals(testdata_pfs_ids_geolife.geometry)).all()
+        pd.testing.assert_frame_equal(pfs.drop('geom', axis=1), testdata_pfs_ids_geolife.drop('geom', axis=1),
+                                      check_like=True)
+
+    def test_tripleg_ids_in_positionfixes_case3(self, geolife_pfs_spts_short, testdata_pfs_ids_geolife):
+        """
+        checks if positionfixes are not altered and if staypoint/tripleg ids where set correctly during tripleg
+        generation using the 'between_staypoints' method with only positionfixes as input (case 3).
+        """
+        pfs, spts = geolife_pfs_spts_short
+        pfs, tpls = pfs.as_positionfixes.generate_triplegs(method='between_staypoints')
+
+        # check if staypoint/tripleg ids are present
+        assert (pfs.geometry.geom_almost_equals(testdata_pfs_ids_geolife.geometry)).all()
+        pd.testing.assert_frame_equal(pfs.drop('geom', axis=1), testdata_pfs_ids_geolife.drop('geom', axis=1),
+                                      check_like=True)
+
+    def test_tripleg_generation_stability(self, geolife_pfs_spts_long):
+        """
+        checks if the results are same if different variants of the tripleg_generation method 'between_staypoints'
+        are used
+        """
+
+        pfs, spts = geolife_pfs_spts_long
+
+        pfs_case1, tpls_case1 = pfs.as_positionfixes.generate_triplegs(spts, method='between_staypoints')
+        pfs_case2, tpls_case2 = pfs.drop('staypoint_id', axis=1).as_positionfixes.generate_triplegs \
+            (spts, method='between_staypoints')
```

Comment:
For me using `\` to split the line reduces the readability. Is this common in the numpy standards?

Reply:
yes, but splitting within parentheses (which is possible here) should be preferred: https://www.python.org/dev/peps/pep-0008/#maximum-line-length